### PR TITLE
improve the consistency of the readme and add links

### PR DESCRIPTION
**Pull request by maasg**
### What changes were proposed in this pull request?

Improves the README.
- Adds the value proposition of at the different stages of the application lifecycle
- Improves organization and coherence of the docs
- Organize the links

### Why are the changes needed?

Improve the documentation to give our users the best introduction possible.

### Does this PR introduce any user-facing change?

Yes - docs.

### How was this patch tested?

I read it again :-)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +7,13 @@ Cloudflow allows you to easily break down your streaming application to smaller
 Cloudflow integrates with popular streaming engines like Akka, Spark and Flink. It also comes with a powerful CLI tool to easily manage, scale and configure your streaming applications at runtime.
 With its powerful abstractions, Cloudflow allows to define, build and deploy the most complex streaming applications.
 
-In a nutshell, Cloudflow is an application development toolkit composed of:
+- Develop: Focus only on business logic, leave the boilerplate to us.
```
Comment (olofwalker) [inline README.md]:
```suggestion
- Develop: Focus only on business logic.
```

Comment (maasg) [reply README.md]: @olofwalker I actually like the original statement. It describes part of what we do as we communicated here: https://www.lightbend.com/blog/cloudflow-released-lightbends-newest-open-source-project
